voltdb/src/test/java/com/yahoo/ycsb/db/voltdb/test/VoltDBClientTest.java

```diff
@@ -0,0 +1,444 @@
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */ 
+
+package com.yahoo.ycsb.db.voltdb.test;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.assumeNoException;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+import com.yahoo.ycsb.db.voltdb.ConnectionHelper;
+import com.yahoo.ycsb.db.voltdb.VoltClient4;
+
+import org.junit.*;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Vector;
+import java.util.Properties;
+
+/**
+ * Test harness for YCSB / VoltDB. Note that not much happens if voltb isn't visible.
+ * 
+ */
+public class VoltDBClientTest {
+
+  private static final String TABLE_NAME = "USERTABLE";
+  private static final int FIELD_LENGTH = 32;
+  private static final String FIELD_PREFIX = "FIELD";
+  private static final int NUM_FIELDS = 3;
+
+  private static final String INSERT_TEST_KEY = "InsertReadTest";
+  private static final String INSERT_DELETE_AND_READ_TEST_KEY = "InsertDeleteReadTest";
+  private static final String UPDATE_TEST_KEY = "UpdateTest";
+  private static final String NON_EXISTENT_KEY = "NonExistTest";
+
+  private static final String SCAN_KEY_PREFIX = "ScanKey_";
+  private static final int SCAN_RECORD_COUNT = 5000;
+
+  private static final String[] TEST_DATA_KEYS = { INSERT_TEST_KEY, INSERT_DELETE_AND_READ_TEST_KEY, UPDATE_TEST_KEY };
+
+  private static VoltClient4 voltClient = null;
+  private static boolean haveDb = false;
+
+  boolean hasVoltDBInstance = false;
+
+  @BeforeClass
+  public static void setup() {
+
+    Properties p = new Properties();
+
+    String servers = p.getProperty("voltdb.servers", "localhost");
+    String user = p.getProperty("voltdb.user", "");
+    String password = p.getProperty("voltdb.password", "");
+    String strLimit = p.getProperty("voltdb.ratelimit", "70000");
+
+    p.setProperty("voltdb.servers", servers);
+    p.setProperty("voltdb.user", user);
+    p.setProperty("voltdb.password", password);
+    p.setProperty("voltdb.ratelimit", strLimit);
+
+    try {
+      voltClient = new VoltClient4();
+      voltClient.setProperties(p);
+
+      if (ConnectionHelper.checkDBServers(servers)) {
+        voltClient.init();
+        haveDb = true;
+        removeExistingData();
+      }
+
+    } catch (Exception e) {
```

Comment:
This probably needs to be wrapped in `assumeNoException` instead of a try/catch so that the test only runs when there's a local VoltDB instance.

Reply:
Done